tests/subworkflows/test_patch.py

```diff
@@ -0,0 +1,307 @@
+from ..utils import GITLAB_REPO, GITLAB_SUBWORKFLOWS_BRANCH, GITLAB_URL
+
+OLD_SHA = "dbb12457e32d3da8eea7dc4ae096201fff4747c5"
+SUCCEED_SHA = "0a33e6a0d730ad22a0ec9f7f9a7540af6e943221"
+FAIL_SHA = "b6e5e8739de9a1a0c4f85267144e43dbaf8f1461"
+
+
+class TestSubworkflowsPatch(TestSubworkflows):
+    """
+    Test the 'nf-core subworkflows patch' command
+    """
+
+    def modify_main_nf(self, path):
+        """Modify a file to test patch creation"""
+        with open(path) as fh:
+            lines = fh.readlines()
+        # We want a patch file that looks something like:
+        # -    ch_fasta // channel: [ fasta ]
+        for line_index in range(len(lines)):
+            if lines[line_index] == "    ch_fasta // channel: [ fasta ]\n":
+                to_pop = line_index
+        lines.pop(to_pop)
+        with open(path, "w") as fh:
+            fh.writelines(lines)
+
+    def setup_patch(self, pipeline_dir, modify_subworkflow):
+        # Install the subworkflow bam_sort_stats_samtools
+        install_obj = nf_core.subworkflows.SubworkflowInstall(
+            pipeline_dir,
+            prompt=False,
+            force=False,
+            remote_url=GITLAB_URL,
+            branch=GITLAB_SUBWORKFLOWS_BRANCH,
+            sha=OLD_SHA,
+        )
+
+        # Install the module
+        install_obj.install("bam_sort_stats_samtools")
+
+        if modify_subworkflow:
+            # Modify the subworkflow
+            subworkflow_path = Path(pipeline_dir, "subworkflows", GITLAB_REPO, "bam_sort_stats_samtools")
+            self.modify_main_nf(subworkflow_path / "main.nf")
+
+    def test_create_patch_no_change(self):
+        """Test creating a patch when there is a change to the module"""
+        self.setup_patch(self.pipeline_dir, False)
+
+        # Try creating a patch file
+        patch_obj = nf_core.subworkflows.SubworkflowPatch(self.pipeline_dir, GITLAB_URL, GITLAB_SUBWORKFLOWS_BRANCH)
+        with pytest.raises(UserWarning):
+            patch_obj.patch("bam_sort_stats_samtools")
+
+        subworkflow_path = Path(self.pipeline_dir, "subworkflows", GITLAB_REPO, "bam_sort_stats_samtools")
+
+        # Check that no patch file has been added to the directory
+        assert set(os.listdir(subworkflow_path)) == {"main.nf", "meta.yml"}
+
+    def test_create_patch_change(self):
+        """Test creating a patch when there is no change to the subworkflow"""
+        self.setup_patch(self.pipeline_dir, True)
+
+        # Try creating a patch file
+        patch_obj = nf_core.subworkflows.SubworkflowPatch(self.pipeline_dir, GITLAB_URL, GITLAB_SUBWORKFLOWS_BRANCH)
+        patch_obj.patch("bam_sort_stats_samtools")
+
+        subworkflow_path = Path(self.pipeline_dir, "subworkflows", GITLAB_REPO, "bam_sort_stats_samtools")
+
+        # Check that a patch file with the correct name has been created
+        assert set(os.listdir(subworkflow_path)) == {"main.nf", "meta.yml", "bam_sort_stats_samtools.diff"}
```

Comment:
less error prone if we only check that `bam_sort_stats_samtools.diff` is in the directory